File: 2020-12-28/BOJ/10952/java/10952-2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.lang.StringBuilder;
import java.util.Arrays;
import java.util.Optional;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        String input = "";
        while ((input = bf.readLine()) != null) {
            Optional<Integer> tmp = Arrays.stream(input.split(" ")).map(i -> Integer.parseInt(i))
                    .reduce((accu, curr) -> {
                        return accu + curr;
                    });
            if (tmp.get() == 0)
                break;
            sb.append(tmp.get() + "\n");
        }
        System.out.println(sb);
    }
}

File: 2020-12-28/BOJ/10952/js/10952-2.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.on("line", (line) => {
  line
    .split(" ")
    .map((e) => Number(e))
    .reduce((accu, curr) => {
      if (accu + curr > 0) process.stdout.write(string(accu + curr) + "\n");
    });
}).on("close", () => process.exit());
